src/content/docs/en/guides/cms/kontent-ai.mdx

```diff
@@ -0,0 +1,514 @@
+
+### Prerequisites
+
+To get started, you'll need the following:
+
+1. **Kontent.ai project** - If you don't have a Kontent.ai account yet, [sign up for free](https://app.kontent.ai/sign-up) and create a new project.
+
+2. **Delivery API keys** - You will need the Environment ID for published content and the Preview API key for fetching drafts (optional). Find them both in **Environment Settings -> API keys** tab in Kontent.ai.
+
+### Setting up credentials
+
+To add your Kontent.ai project credentials to Astro, create a `.env` file in the root of your project with the following variables:
+
+```ini title=".env"
+KONTENT_ENVIRONMENT_ID=YOUR_ENVIRONMENT_ID
+KONTENT_PREVIEW_API_KEY=YOUR_PREVIEW_API_KEY
+```
+
+Now, you can use these environment variables in your Astro project.
+
+### Installing dependencies
+
+To connect Astro with your Kontent.ai project, install the [Kontent.ai TypeScript SDK](https://github.com/kontent-ai/delivery-sdk-js):
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+    npm install @kontent-ai/delivery-sdk
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+    pnpm install @kontent-ai/delivery-sdk
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+    yarn add @kontent-ai/delivery-sdk
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+Next, create a new file called `kontent.ts` in the `src/lib/` directory of your Astro project.
+
+```ts title="src/lib/kontent.ts"
+import { createDeliveryClient } from "@kontent-ai/delivery-sdk";
+
+export const deliveryClient = createDeliveryClient({
+    environmentId: import.meta.env.KONTENT_ENVIRONMENT_ID,
+    previewApiKey: import.meta.env.KONTENT_PREVIEW_API_KEY,
+});
+```
+
+This implementation creates a new `DeliveryClient` using credentials from the `.env` file.
+
+*Note: The `previewApiKey` is optional. When used, you can [configure each query](https://github.com/kontent-ai/delivery-sdk-js#enable-preview-mode-per-query) to the Delivery API endpoint to return the latest versions of content items regardless of their state in workflow. Otherwise, only published items are returned.*
+
+Finally, your root directory should now include these new files:
+
+<FileTree title="Project Structure">
+- src/
+  - lib/
+    - **kontent.ts**
+- **.env**
+- astro.config.mjs
+- package.json
+</FileTree>
+
+### Fetching data
+
+The `DeliveryClient` is now available to all components. To fetch content, use the `DeliveryClient` and fluent API syntax to define your desired items. This example shows a basic fetch of blog posts and renders their titles in a list:
+
+```astro
+---
+import { deliveryClient } from "../lib/kontent";
+
+const blogPosts = await deliveryClient
+    .items()
+    .type("blogPost")
+    .toPromise()
+---
+<ul>
+  {blogPosts.data.items.map(blogPost => (
+    <li>{blogPost.elements.title.value}</li>
+  ))}
+</ul>
+```
```

Comment:
maybe add a title and highlighted lines here

Reply:
Added